Enforcer rule not working in Windows? Because of linkage errors?

[log](https://fusion.corp.google.com/runanalysis/buildlogs/prod%3Acloud-opensource-java%2Fgcp_windows%2Fpresubmit/prod%3Acloud-opensource-java%2Fgcp_windows%2Fpresubmit/KOKORO/b2ec87ca-a246-4e68-b17e-2409870e2037/0/prod%3Acloud-opensource-java%2Fgcp_windows%2Fpresubmit/Targets)

```
       |        [INFO] ------------------------------------------------------------------------
       |        [INFO] BUILD FAILURE
       |        [INFO] ------------------------------------------------------------------------
       |        [INFO] Total time: 14.093 s
       |        [INFO] Finished at: 2020-04-17T12:14:40-07:00
       |        [INFO] Final Memory: 19M/285M
       |        [INFO] ------------------------------------------------------------------------
       |        [ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce (enforce-linkage-checker) on project no-such-method-error-example: Execution enforce-linkage-checker of goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce failed: An API incompatibility was encountered while executing org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce: java.lang.NoSuchMethodError: org.eclipse.aether.transfer.TransferResource.<init>(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/io/File;Lorg/eclipse/aether/RequestTrace;)V
       |        [ERROR] -----------------------------------------------------
       |        [ERROR] realm =    plugin>org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3
       |        [ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
       |        [ERROR] urls[0] = file:/T:/src/github/cloud-opensource-java/enforcer-rules/target/local-repo/org/apache/maven/plugins/maven-enforcer-plugin/3.0.0-M3/maven-enforcer-plugin-3.0.0-M3.jar
       |        [ERROR] urls[1] = file:/T:/src/github/cloud-opensource-java/enforcer-rules/target/local-repo/com/google/cloud/tools/linkage-checker-enforcer-rules/1.2.2-SNAPSHOT/linkage-checker-enforcer-rules-1.2.2-SNAPSHOT.jar
```


```
java.lang.NoSuchMethodError:
org.eclipse.aether.transfer.TransferResource.<init>
(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;Ljava/io/File;Lorg/eclipse/aether/RequestTrace;)V
```

The method has been there since maven-resolver-api:1.1.0:

```
    /**
     * Creates a new transfer resource with the specified properties.
     *
     * @param repositoryId The ID of the repository used to transfer the resource, may be {@code null} or
     *                     empty if unknown.
     * @param repositoryUrl The base URL of the repository, may be {@code null} or empty if unknown. If not empty, a
     *            trailing slash will automatically be added if missing.
     * @param resourceName The relative path to the resource within the repository, may be {@code null}. A leading slash
     *            (if any) will be automatically removed.
     * @param file The source/target file involved in the transfer, may be {@code null}.
     * @param trace The trace information, may be {@code null}.
     *
     * @since 1.1.0
     */
    public TransferResource( String repositoryId, String repositoryUrl, String resourceName,
        File file, RequestTrace trace )
...
```


The solution of this issue be related https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1091.

--- a/.mvn/wrapper/maven-wrapper.properties
+++ b/.mvn/wrapper/maven-wrapper.properties
@@ -1,2 +1,2 @@
-distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.6.3/apache-maven-3.6.3-bin.zip
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.6.0/apache-maven-3.6.0-bin.zip
 wrapperUrl=https://repo.maven.apache.org/maven2/io/takari/maven-wrapper/0.5.6/maven-wrapper-0.5.6.jar

